File: WEB-INF/classes/pawfect_home/mail.py
from email.message import EmailMessage
import ssl
import smtplib
import sys
import os

class Mail:
    def __init__(self, recipient, body):
        self.subject = 'A person is interested in your pet!'
        self.body = body
        self.sender = 'pawfecthome22@gmail.com'
        self.recipient = recipient
        self.password = os.environ['EMAIL_PASSWORD']

    def send_email(self):
        em = EmailMessage()
        em['From'] = mail.sender
        em['To'] = mail.recipient
        em['Subject'] = mail.subject
        em.set_content(mail.body)
        context = ssl.create_default_context()
        with smtplib.SMTP_SSL('smtp.gmail.com', 465, context= context) as smtp:
            smtp.login(mail.sender,mail.password)
            smtp.sendmail(mail.sender, mail.recipient, em.as_string())

body = (sys.argv[1])
recipient=(sys.argv[2])
mail=Mail(recipient, body)
mail.send_email()

File: WEB-INF/classes/pawfect_home/Mail.java
package pawfect_home;
import java.util.Properties;
import javax.mail.*;
import javax.mail.internet.InternetAddress;
import javax.mail.internet.MimeMessage;

public class Mail {
    private String subject;
    private String body;
    private String sender;
    private String recipient;
    private String password;

    public Mail(String recipient, String body) {
        this.subject = "A person is interested in your pet!";
        this.body = body;
        this.sender = "pawfecthome22@gmail.com";
        this.recipient = recipient;
        this.password = System.getenv("EMAIL_PASSWORD");
    }

    public void sendEmail() {
        Properties props = new Properties();
        props.put("mail.smtp.host", "smtp.gmail.com");
        props.put("mail.smtp.port", "587");
        props.put("mail.smtp.auth", "true");
        props.put("mail.smtp.starttls.enable", "true");

        Session session = Session.getInstance(props, new Authenticator() {
            protected PasswordAuthentication getPasswordAuthentication() {
                return new PasswordAuthentication(sender, password);
            }
        });

        try {
            MimeMessage message = new MimeMessage(session);
            message.setFrom(new InternetAddress(sender));
            message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(recipient));
            message.setSubject(subject,"UTF-8");
            message.setText(body, "text/html; charset=UTF-8");
            message.setContent(body, "text/html; charset=UTF-8");
            Transport.send(message);
            Transport.send(message);
        } catch (MessagingException e) {
            e.printStackTrace();
        }
    }
}
